File: backend/src/main/java/com/example/backend/entity/Project.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "projects")
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    String name;

    String description;

    @Column(name = "avatar_url")
    String avatarUrl = "https://images.icon-icons.com/2699/PNG/512/atlassian_jira_logo_icon_170511.png";

    String key;

    @Column(name = "is_public")
    Boolean isPublic = false;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    OffsetDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "team_id")
    Team team;

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "chat_id")
    Chat chat;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    User creator;

    @JsonIgnore
    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
    List<Issue> issues=new ArrayList<>();

    @JsonIgnore
    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
    List<ProjectMember> projectMembers = new ArrayList<>();
}

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description: string;
  avatarUrl: string | null;
  key: string;
  isPublic?: boolean;
  createdAt: string;
  updatedAt: string;
  teamId: string;
  chatId: string;
  creator: {
    id: string;
    email: string;
    name: string;
    avatarUrl: string;
    provider: string;
    createdAt: string;
    updatedAt: string;
  };
  members: ProjectMember[];
  memberCount: number;
}

export interface ProjectCreationRequest {
  name: string;
  description: string;
  key: string;
  teamId: string;
  avatarUrl: string;
  isPublic: boolean;
}

export interface ProjectUpdateRequest {
  name?: string;
  description?: string;
}

export interface ProjectMember {
  projectId: string;
  userId: string;
  role: 'ADMIN' | 'MEMBER' | 'PM';
  joinedAt: string;
  user: {
    id: string;
    email: string;
    name: string;
    avatarUrl: string;
    provider: string;
    createdAt: string;
    updatedAt: string;
  };
}

export interface ProjectResponse {
  code: number;
  message: string;
  result: Project;
}

export interface ProjectsResponse {
  code: number;
  message: string;
  result: Project[];
} 